`type.argument.inference.crashed` error with method reference

Input:
```java
import java.util.*;
import java.util.stream.*;
abstract class BlockTest {
  Set<BasicBlock<?>> iteratedBlocks;
  String test() {
    Collection<? extends BasicBlock<?>> blocks = getBlocks();
    return blocks.stream()
        .filter(n -> !iteratedBlocks.contains(n))
        .map(BasicBlock::getStmts)
        .collect(Collectors.toList())
        .toString();
  }
  interface BasicBlock<V extends BasicBlock<V>> {
    List<Stmt> getStmts();
  }
  interface Stmt {}
  abstract Collection<? extends BasicBlock<?>> getBlocks();
}
```
I get a `type.argument.inference.crashed` error with the Tainting Checker:
```
$ checker-framework/checker/bin/javac -processor tainting BlockTest.java
BlockTest.java:12: error: [type.argument.inference.crashed] Type argument inference crashed for BasicBlock.getStmts
                  .map(BasicBlock::getStmts)
                       ^
  error: An exception occurred:  False bound for: Constraint: ? extends @Tainted Object <= use of a2 Result: FALSE
1 error
```
We see a similar error with the Resource Leak Checker.

--- a/framework/src/main/java/org/checkerframework/framework/util/typeinference8/constraint/Typing.java
+++ b/framework/src/main/java/org/checkerframework/framework/util/typeinference8/constraint/Typing.java
@@ -211,7 +211,9 @@ public class Typing extends TypeConstraint {
       // constraint reduces to the following new constraints:
       // for all i (1 <= i <= n), <Bi <= Ai>.
 
-      AbstractType sAsSuper = S.asSuper(T.getJavaType());
+      // Capturing is not in the JLS, but otherwise wildcards appear in the constraints against
+      // the type arguments, which causes crashes.
+      AbstractType sAsSuper = S.asSuper(T.getJavaType()).capture(context);
       if (sAsSuper == null) {
         return ConstraintSet.FALSE;
       } else if (sAsSuper.isRaw() || T.isRaw()) {
@@ -300,11 +302,9 @@ public class Typing extends TypeConstraint {
    */
   private ReductionResult reduceContained() {
     if (T.getTypeKind() != TypeKind.WILDCARD) {
-      // The JLS says that if S is a wildcard the constraint should reduce to false,
-      // but javac seems to accept this case. Issue6725.java is an example.
-      // if (S.getTypeKind() == TypeKind.WILDCARD) {
-      //   return ConstraintSet.FALSE;
-      // }
+      if (S.getTypeKind() == TypeKind.WILDCARD) {
+        return ConstraintSet.FALSE;
+      }
       if (isCovarTypeArg) {
         return new Typing(S, T, Kind.SUBTYPE);
       }

--- a/framework/tests/all-systems/Issue6769.java
+++ b/framework/tests/all-systems/Issue6769.java
@@ -1,0 +1,14 @@
+package typearginfer;
+
+import java.util.*;
+import java.util.stream.*;
+
+abstract class Issue6769 {
+  void test(Stream<? extends BasicBlock<?>> blocks) {
+    blocks.map(BasicBlock::getStmts);
+  }
+
+  interface BasicBlock<V extends BasicBlock<V>> {
+    List<String> getStmts();
+  }
+}

